dccrg.hpp

```diff
@@ -8993,6 +8995,18 @@ template <
 								return true;
 							}
 
+							// Instead of splitting paths, check whether our target cell is among the siblings at same transverse indices, i.e. on the same face seen from one level coarser.
+							const uint64_t parent = this->get_parent(cellHere);
+							const std::vector<uint64_t> siblings = this->get_all_children(parent);
+							for(uint64_t sibling : siblings) {
+								if(sibling == cell2) {
+									const Types<3>::indices_t sibling_indices = this->mapping.get_indices(sibling);
+									if(sibling_indices[(dimension+1)%3] == indices2[(dimension+1)%3] && sibling_indices[(dimension+2)%3] == indices2[(dimension+2)%3]) {
+										return true;
+									}
+								}
+							}
+
```

Comment:
I'm not sure I agree with the logic here.
We are anyway doing our stepping logically on the finest level, so why would we need to care about the actual refinement level here? If we have stepped on the finest level and reached somewhere that is *not* our target cell, why still look at its siblings?

Reply:
Note that we can *not* assume the basic requirement of "only one refinement step in the neighborhood" holds here, because this function gets used specifically to determine which cells would need to be refined in order to fulfil it. So it might have multiple steps at the moment it runs.

Reply:
We don't "catch" all possible eventual neighbours on a finer level, as I figured out with cerr-debugging, as we don't split paths. Or at least, by checking whether one of the target cell's siblings is our target and accepting that, I got rid of the consistency complaints of the -DDEBUG mode. If you think I diagnosed that wrongly, I am happy to share my spreadsheet and let you have a go. :)

Reply:
If we need to account for more-than-one we might want to keep track of the refinement level difference between source and target and search for corresponding siblings or cousins? I guess this should anyway only be performed if we go from coarse to fine? Is that why I get the weird meshes out of my last test on Zulip?